this currently fails the new `rpc_dumptxoutset.py` test, both in CI and locally
```
2019-10-24T07:46:53.097000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20191024_094030/rpc_dumptxoutset_22
2019-10-24T07:46:53.489000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):                      
  File "/store/orion/projects/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 194, in main
    self.run_test()                                      
  File "/home/orion/projects/bitcoin/bitcoin/test/functional/rpc_dumptxoutset.py", line 39, in run_test
    '10c25df662424c94d06458aa5ecc7bd73680aa098f5093fdd5fa27fb2af08568')
  File "/store/orion/projects/bitcoin/bitcoin/test/functional/test_framework/util.py", line 46, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(4fac28ba32694cf370432ca408620e053a370f11de8b19488568ed1308d9b511 == 10c25df662424c94d06458aa5ecc7bd73680aa098f5093fdd5fa27fb2af08568)
2019-10-24T07:46:53.565000Z TestFramework (INFO): Stopping nodes
2019-10-24T07:46:53.817000Z TestFramework (WARNING): Not cleaning up dir /tmp/test_runner_₿_🏃_20191024_094030/rpc_dumptxoutset_22
2019-10-24T07:46:53.818000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/test_runner_₿_🏃_20191024_094030/rpc_dumptxoutset_22/test_framework.log
2019-10-24T07:46:53.819000Z TestFramework (ERROR): Hint: Call /store/orion/projects/bitcoin/bitcoin/test/functional/combine_logs.py '/tmp/test_runner_₿_🏃_20191024_094030/rpc_dumptxoutset_22' to cons
olidate all logs  
```

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -159,6 +159,7 @@ BITCOIN_CORE_H = \
   node/coinstats.h \
   node/psbt.h \
   node/transaction.h \
+  node/utxo_snapshot.h \
   noui.h \
   optional.h \
   outputtype.h \

--- a/src/node/utxo_snapshot.h
+++ b/src/node/utxo_snapshot.h
@@ -1,0 +1,50 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_NODE_UTXO_SNAPSHOT_H
+#define BITCOIN_NODE_UTXO_SNAPSHOT_H
+
+#include <uint256.h>
+#include <serialize.h>
+
+//! Metadata describing a serialized version of a UTXO set from which an
+//! assumeutxo CChainState can be constructed.
+class SnapshotMetadata
+{
+public:
+    //! The hash of the block that reflects the tip of the chain for the
+    //! UTXO set contained in this snapshot.
+    uint256 m_base_blockhash;
+
+    //! The number of coins in the UTXO set contained in this snapshot. Used
+    //! during snapshot load to estimate progress of UTXO set reconstruction.
+    uint64_t m_coins_count = 0;
+
+    //! Necessary to "fake" the base nChainTx so that we can estimate progress during
+    //! initial block download for the assumeutxo chainstate.
+    unsigned int m_nchaintx = 0;
+
+    SnapshotMetadata() { }
+    SnapshotMetadata(
+        const uint256& base_blockhash,
+        uint64_t coins_count,
+        unsigned int nchaintx) :
+            m_base_blockhash(base_blockhash),
+            m_coins_count(coins_count),
+            m_nchaintx(nchaintx) { }
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITE(m_base_blockhash);
+        READWRITE(m_coins_count);
+        READWRITE(m_nchaintx);
+    }
+
+};
+
+#endif // BITCOIN_NODE_UTXO_SNAPSHOT_H

--- a/src/validation.h
+++ b/src/validation.h
@@ -21,6 +21,7 @@
 #include <txmempool.h> // For CTxMemPool::cs
 #include <txdb.h>
 #include <versionbits.h>
+#include <serialize.h>
 
 #include <algorithm>
 #include <atomic>

